nix/default.nix

```
@@ -38,6 +40,13 @@ let
       cd $out
       cp ${packageLock} ./package-lock.json
       cp ${packageJson} ./package.json
+      cp ${
+        pkgs.linkFarm "node-packages-${projectName}-source"
+          (builtins.map (path: {
+            inherit path;
+            name = builtins.baseNameOf (builtins.toString path);
+          }) submodules)
+      }/* . -r
```

Here's a patch for what @klntsky suggested.

```suggestion
      ${pkgs.lib.optionalString (builtins.length submodules != 0) "cp ${
        pkgs.linkFarm "node-packages-${projectName}-source"
          (builtins.map (path: {
            inherit path;
            name = builtins.baseNameOf (builtins.toString path);
          }) submodules)
      }/* . -r"}
```